src/object/obj_tx.c

```diff
@@ -1218,6 +1222,11 @@ dc_tx_classify_common(struct dc_tx *tx, struct daos_cpd_sub_req *dcsr,
 			  shard->do_target_id, dtrg_nr);
 
 		dtrg = &dtrgs[shard->do_target_id];
+
+		dtrg->dtrg_flags |= tgt_flags;
+		if (unlikely(shard->do_shard != idx))
```

Comment:
I don't understand the related code, just confirm here the "do_shard != idx" is which case? OSA added temporary shard? thx

Reply:
Yes, it is for OSA case.